docs/source/user_guides/optimization_guide.rst

```diff
@@ -0,0 +1,164 @@
+* Using the Streamlit App: A Streamlit app is available for viewing and comparing benchmark results. You can upload benchmark result files using the app's "Benchmark sets" tab, inspect individual results in the "View" tab, and compare metrics in the "Compare" tab.
+
+**Additional Information**
+
+DFF provides configuration presets in the `basic config <../apiref/dff.utils.db_benchmarks.basic_config.py>`__ module,
+covering various contexts, messages, and edge cases.
+You can use these presets by passing them to the benchmark functions.
+
+If the basic configurations are not sufficient for your needs, you can create custom configurations by inheriting from the BenchmarkConfig class.
+You'll need to define three methods:
+
+* `get_context` for getting initial contexts,
+* `info` for getting display information representing the configuration,
+* and `context_updater` for updating contexts.
+
+.. note::
+
+    The exact instructions of how the testing can be carried out are available in the
+    `DB benchmarking tutorial <../tutorials/tutorials.context_storages.8_db_benchmarking.py>`__.
+
+Optimization techniques
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Aside from choosing an appropriate database type, there exists a number of other recommendations
+that may help you improve the efficiency of your service.
+
+* Firstly, follow the DRY principle not only with regard to your code, but also with regard to
+  computational operations. In other words, you have to make sure that your callback functions work only once
+  during a dialog turn and only when needed. E.g. you can take note of the `conditions` api available as a part
+  of the `Pipeline` module: while normally a pipeline service runs every turn, you can restrict it
+  to only run on turns when a particular condition is satisfied, greatly reducing
+  the number of performed actions (see the
+  `Groups and Conditions tutorial <../tutorials/tutorials.pipeline.4_groups_and_conditions_full.py>`__).
+
+* Using caching for resource-consuming callbacks and actions may also prove to be a helpful strategy.
+  In this manner, you can improve the computational efficiency of your pipeline,
+  while making very few changes to the code itself. DFF includes a caching mechanism
+  for response functions. However, the simplicity
+  of the DFF API makes it easy to integrate any custom caching solutions that you may come up with.
+  See the `Cache tutorial <../tutorials/tutorials.utils.1_cache.py>`__.
+
+* Finally, be mindful about the use of computationally expensive algorithms, like NLU classifiers
+  or LLM-based generative networks, since those require a great deal of time and resources
+  to produce an answer. In case you need to use one, take full advantage of caching along with
+  other means to relieve the computational load imposed by neural networks such as message queueing.
+
+..
+    todo: add a link to a user guide about using message queueing.
```

Comment:
Or maybe we should merge that guide (see `[D] User Guide about Side Features` https://github.com/deeppavlov/dialog_flow_framework/discussions/237) with this one.

_I left this todo, fyi_

Reply:
Maybe, we can add the link in the same PR with the side features guide

Reply:
Yeah, sure. That's what the todo is for.